File: Problem-030/solution.py
def digit_power_sum(n, power):
    total = 0

    for digit in str(n):
        total += int(digit) ** power

    return total


def digit_fifth_powers(power):

    total = 0

    for n in range(2, 1000001):

        if digit_power_sum(n, power) == n:
            total += n

    return total

File: Problem-030/solution.js
function digitnPowers(n) {
    let total = 0;

    for (let num = 2; num <= 1000000; num++) {

        let sum = 0;

        for (let digit of num.toString()) {
            sum += Number(digit) ** n;
        }

        if (sum === num) {
            total += num;
        }
    }

    return total;
}
